carDevise/app/controllers/cars_controller.rb

```diff
@@ -0,0 +1,76 @@
+class CarsController < ApplicationController
+  before_action :set_car, only: [:show ]
+  before_action :authenticate_user! [:show, :edit, :update, :destroy]
+
+  # GET /cars
+  # GET /cars.json
+  def index
+    @cars = Car.all
+  end
+
+  # GET /cars/1
+  # GET /cars/1.json
+  def show
+  end
+
+  # GET /cars/new
+  def new
+    @car = current_user.cars.new
+  end
+
+  # GET /cars/1/edit
+  def edit
+    @car = current_user.cars.find(params[:id])
+  end
+
+  # POST /cars
+  # POST /cars.json
+  def create
+    @car = current_user.cars.new(car_params)
+
+    respond_to do |format|
+      if @car.save
+        format.html { redirect_to @car, notice: 'Car was successfully created.' }
+        format.json { render :show, status: :created, location: @car }
+      else
+        format.html { render :new }
+        format.json { render json: @car.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /cars/1
+  # PATCH/PUT /cars/1.json
+  def update
+    respond_to do |format|
+      if @car.update(car_params)
+        format.html { redirect_to @car, notice: 'Car was successfully updated.' }
+        format.json { render :show, status: :ok, location: @car }
+      else
+        format.html { render :edit }
+        format.json { render json: @car.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /cars/1
+  # DELETE /cars/1.json
+  def destroy
+    @car.destroy
```

Comment:
Good work! but @car is not declared so you have first to find the car that user want to destroy
```rb
def destroy
    @car = current_user.cars.find(params[:id])
    @car.destroy
    respond_to do |format|
      format.html { redirect_to cars_url, notice: 'Car was successfully destroyed.' }
      format.json { head :no_content }
    end
  end
```